language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/document/BallerinaDocumentService.java

```diff
@@ -289,14 +296,104 @@ public CompletableFuture<BallerinaSyntaxTreeResponse> triggerModify(BallerinaTri
         return CompletableFuture.supplyAsync(() -> reply);
     }
 
+    @JsonRequest
+    public CompletableFuture<BallerinaSyntaxTreeResponse> syntaxTreeByName(BallerinaSyntaxTreeByNameRequest request) {
+        return CompletableFuture.supplyAsync(() -> {
+            AtomicReference<String> sourceCode = new AtomicReference<>();
+            AtomicReference<JsonElement> jsonSyntaxTree = new AtomicReference<>();
+            AtomicReference<String> defFilePath = new AtomicReference<>();
+            BallerinaSyntaxTreeResponse reply = new BallerinaSyntaxTreeResponse();
+            String fileUri = request.getDocumentIdentifier().getUri();
+            Optional<Path> filePath = PathUtil.getPathFromURI(fileUri);
+            if (filePath.isEmpty()) {
+                return reply;
+            }
+
+            try {
+                Optional<Document> srcFile = this.workspaceManagerProxy.get().document(filePath.get());
+                if (srcFile.isEmpty()) {
+                    return reply;
+                }
+
+                // Get the semantic model.
+                Optional<SemanticModel> semanticModel = this.workspaceManagerProxy.get().semanticModel(filePath.get());
+
+                // Get the start line range of function invoke.
+                int lineValue = request.getLineRange().getStart().getLine();
+                int charValue = request.getLineRange().getStart().getCharacter();
+
+                // Get the symbol of function
+                Optional<Symbol> functionSymbol = semanticModel.get().symbol(srcFile.get(),
+                        LinePosition.from(lineValue, charValue));
+
+                // Get the file path of the function symbol
+                String functionPath = functionSymbol.get().getLocation().get().lineRange().filePath();
```

Comment:
Shall we add a recovery logic(or throw a proper exception) to avoid `NoSuchElementException`s incase the `functionSymbol` is empty?